gbpservice/neutron/nsf/core/main.py

```diff
@@ -0,0 +1,705 @@
+                % (identify(eh),
+                    ev.identify(), t.identify())))
+
+    def run(self, qu):
+        LOG.info(_("Started worker process - %s" % (process)))
+        while True:
+            ev = self._get()
+            if ev:
+                LOG.debug(_("Got event %s" % (ev.identify())))
+                eh = self._eh.get(ev)
+                if not ev.poll_event:
+                    t = self._tpool.dispatch(eh.handle_event, ev)
+                    LOG.debug(_(
+                        "Event %s is not poll event - "
+                        "disptaching handle_event() of handler %s"
+                        "to thread %s"
+                        % (ev.identify(), identify(eh), t.identify())))
+                else:
+                    if ev.poll_event == 'POLL_EVENT_CANCELLED':
+                        LOG.debug(
+                            _("Got cancelled event %s" % (ev.identify())))
+                        self._cancelled(eh, ev)
+                    else:
+                        LOG.info(
+                            _("Got POLL Event %s scheduling"
+                                % (ev.identify())))
+                        self._sched(eh, ev)
+            time.sleep(0)  # Yield the CPU
+
+
+class EventHandlers(object):
+
+    def __init__(self):
+        self._ehs = {}
+
+    def register(self, ev):
+        LOG.debug(_("Registering handler %s" % (self.identify(ev))))
+        if ev.id in self._ehs.keys():
+            self._ehs[ev.id].extend([ev])
+        else:
+            self._ehs[ev.id] = [ev]
+
+    def get(self, ev):
+        for id, eh in self._ehs.iteritems():
+            if id == ev.id:
+                LOG.debug(_("Returning handler %s" % (self.identify(eh[0]))))
+                return eh[0].handler
+        return None
+
+    def identify(self, ev):
+        return "%s - %s" % (ev.identify(), identify(ev.handler))
+
+
+class ServiceController(object):
+
+    def __init__(self, conf, modules):
+        self._conf = conf
+        self.modules = modules
+        self._lock = Lock()
+        self._serializer = Serializer(self)
+
+    def lock(self):
+        self._lock.acquire()
+        LOG.debug(_("Acquired lock.."))
+
+    def unlock(self):
+        self._lock.release()
+        LOG.debug(_("Released lock.."))
+
+    def event_done(self, ev):
+        LOG.info(_("Event %s done" % (ev.identify())))
+        mapp = self._serializer.copy()
+        mapp = mapp[ev.worker_attached]
+        if ev.binding_key not in mapp:
+            return
+
+        LOG.debug(_("Checking if event %s in serialize Q"
+                    % (ev.identify())))
+        qu = mapp[ev.binding_key]['queue']
+        for elem in qu:
+            if elem.key == ev.key:
+                LOG.debug(_("Removing event %s from serialize Q"
+                            % (elem.identify())))
+                self._serializer.remove(elem)
+                break
+        self._serializer.deserialize(ev)
+
+    def serialize(self, ev):
```

Comment:
We have classes with those names.. so not to get confused named them like that. Since there are local variables, should be ok I guess.